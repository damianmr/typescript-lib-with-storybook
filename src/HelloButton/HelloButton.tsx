import React, { useState } from 'react';

type HelloButtonProps = {
  text: string;
  onClick: (clickedCount: number) => void;
  className?: string;
};

function addOne(n: number): number {
  return n + 1;
}

export const HelloButton = ({ text, onClick, className }: HelloButtonProps) => {
  const [count, setCount] = useState(0);
  return (
    <button
      className={className}
      onClick={() => {
        onClick(addOne(count));
        setCount(addOne(count));
      }}
    >
      {text}
    </button>
  );
};
