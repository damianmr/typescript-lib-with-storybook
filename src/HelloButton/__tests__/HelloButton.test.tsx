import React from 'react';
import { HelloButton } from '../HelloButton';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('Renders a HelloButton without crashing.', () => {
  const component = renderer.create(
    <HelloButton className="testClassName" onClick={() => {}} text="Hello" />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
