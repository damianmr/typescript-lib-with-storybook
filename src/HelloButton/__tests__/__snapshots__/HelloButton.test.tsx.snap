// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a HelloButton without crashing. 1`] = `
<button
  className="testClassName"
  onClick={[Function]}
>
  Hello
</button>
`;
